server/src/repositories/tag.repository.ts

```diff
@@ -174,6 +179,34 @@ export class TagRepository implements ITagRepository {
     });
   }
 
+  async deleteEmptyTags() {
+    await this.dataSource.transaction(async (manager) => {
+      const ids = new Set<string>();
+      const tags = await manager.find(TagEntity);
+      for (const tag of tags) {
+        const count = await manager
+          .createQueryBuilder('assets', 'asset')
+          .innerJoin(
+            'asset.tags',
+            'asset_tags',
+            'asset_tags.id IN (SELECT id_descendant FROM tags_closure WHERE id_ancestor = :tagId)',
+            { tagId: tag.id },
+          )
+          .getCount();
+
+        if (count === 0) {
+          this.logger.debug(`Found empty tag: ${tag.id} - ${tag.value}`);
+          ids.add(tag.id);
+        }
+      }
+
+      if (ids.size > 0) {
+        await manager.delete(TagEntity, { id: In([...ids]) });
+        this.logger.log(`Deleted ${ids.size} empty tags`);
+      }
+    });
```

Comment:
```suggestion
    await this.dataSource.query(`
      delete from tags
      where not exists(
        select 1
        from tag_asset
        where tag_asset."tagsId" = tags.id
      )`);
```

Reply:
If I have a tag Parent/Child I don't want to delete Parent or Child if Child is connected to an asset. Does this handle that case?

Reply:
Yes, assuming that if the child has a row in tag_asset, the parent also has a row.

Reply:
The parent doesn't necessarily have a row.

Reply:
Hm, why not? It isn't a parent/child relationship if the parent doesn't always include the child.

Reply:
I don't know how to exactly write the not exists query with the current query builder, but I can try to take a look at this in the future.